### ISSUE TITLE
[Feature Request] Gradient Clipping

### ISSUE THREAD

**Issue by hglee98** [Nota-NetsPresso/netspresso-trainer#505]
### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues.

### Is there any feature that you would like to add?

As mentioned in the title above, it would be beneficial to support gradient clipping features, which are techniques used to mitigate exploding gradients, especially during training. This feature would enhance our training pipeline. Additionally, I can contribute to this issue, which relates to issue #480, as [the official codebase](https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_pytorch/configs/rtdetr/include/optimizer.yml) is also using this feature.

### Is there any suggestion (or solution) to solve this issue?

- Add configuration option to toggle the feature.
- Add Gradient Clipping feature with [torch amp](https://pytorch.org/docs/master/notes/amp_examples.html#gradient-clipping) to the `TrainingPipeline`
- Add documentation 

### Will you help us by submitting a PR?

- [X] I will submit a PR about this issue.
